app/controllers/DemoHttpActionAdapter.java

```diff
@@ -1,22 +1,21 @@
 package controllers;
 
 import org.pac4j.core.context.HttpConstants;
-import org.pac4j.play.PlayWebContext;
+import org.pac4j.core.context.WebContext;
 import org.pac4j.play.http.DefaultHttpActionAdapter;
-import play.mvc.Result;
 
 import static play.mvc.Results.*;
 
 public class DemoHttpActionAdapter extends DefaultHttpActionAdapter {
 
     @Override
-    public Result handle(int code, PlayWebContext context) {
+    public Object adapt(int code, WebContext context) {
```

Comment:
@mmoayyed : here is a specific implementation of the `HttpActionAdapter` where a 401 / 403 error is handled with specific pages.